src/VisualStudio/Core/Def/CallHierarchy/CallHierarchyCommandHandler.cs

```diff
@@ -39,44 +44,58 @@ internal class CallHierarchyCommandHandler : ICommandHandler<ViewCallHierarchyCo
         [SuppressMessage("RoslynDiagnosticsReliability", "RS0033:Importing constructor should be [Obsolete]", Justification = "Used in test code: https://github.com/dotnet/roslyn/issues/42814")]
         public CallHierarchyCommandHandler(
             IThreadingContext threadingContext,
+            IUIThreadOperationExecutor threadOperationExecutor,
             [ImportMany] IEnumerable<ICallHierarchyPresenter> presenters,
             CallHierarchyProvider provider)
         {
             _threadingContext = threadingContext;
+            _threadOperationExecutor = threadOperationExecutor;
             _presenter = presenters.FirstOrDefault();
             _provider = provider;
         }
 
         public bool ExecuteCommand(ViewCallHierarchyCommandArgs args, CommandExecutionContext context)
         {
-            using (var waitScope = context.OperationContext.AddScope(allowCancellation: true, EditorFeaturesResources.Computing_Call_Hierarchy_Information))
+            var document = args.SubjectBuffer.CurrentSnapshot.GetFullyLoadedOpenDocumentInCurrentContextWithChanges(
+                context.OperationContext, _threadingContext);
+            if (document == null)
             {
-                var cancellationToken = context.OperationContext.UserCancellationToken;
-                var document = args.SubjectBuffer.CurrentSnapshot.GetFullyLoadedOpenDocumentInCurrentContextWithChanges(
-                    context.OperationContext, _threadingContext);
-                if (document == null)
-                {
-                    return true;
-                }
+                return true;
+            }
+
+            // We're showing our own UI, ensure the editor doesn't show anything itself.
+            context.OperationContext.TakeOwnership();
+            ExecuteCommandAsync(document, args)
+                .ReportNonFatalErrorAsync();
+
+            return true;
+        }
 
-                var semanticModel = document.GetSemanticModelAsync(cancellationToken).WaitAndGetResult(cancellationToken);
+        private async Task<bool> ExecuteCommandAsync(Document document, ViewCallHierarchyCommandArgs args)
+        {
+            using (var context = _threadOperationExecutor.BeginExecute(
+                ServicesVSResources.Call_Hierarchy, ServicesVSResources.Navigating, allowCancellation: true, showProgress: false))
+            {
+                var cancellationToken = context.UserCancellationToken;
 
+                var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
                 var caretPosition = args.TextView.Caret.Position.BufferPosition.Position;
```

Comment:
do this in the calling method.  so it's guaranteed to be on the UI thread.  pass that value into this async method.  you wan tto avoid touching UI affinitized stuff here as it can be difficult to get it right.